### PR TITLE
Added classes and interfaces for sharepoint

### PR DESCRIPTION
**Pull request by betlgtu**
Added classes and interfaces from
1) cui.js
2) sp.ribbon.js
3) init.js

Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.microsoft.com/en-us/previous-versions/office/developer/sharepoint-2010/ee549700(v=office.14), https://docs.microsoft.com/en-us/previous-versions/office/developer/sharepoint-2010/ff408756(v=office.14)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by andrewbranch** [comment]
This needs to be reviewed by an owner. Thanks for your patience with the holidays.

**Comment by armanio123** [comment]
Pinging owners for review @gandjustas @andrei-markeev @baywet @teroarvola @dennispg @SPWizard01

**Comment by elibarzilay** [comment]
Re-ping @gandjustas @andrei-markeev @baywet @teroarvola @dennispg @SPWizard01

**Review by baywet — CHANGES_REQUESTED** [review]
Hi everyone,
Due to a recent role change I can't maintain the definitions anymore and don't feel I should approve this lengthy PR.
One thing for sure, the addition of static in ClientConstants might break people already referencing it.
This should probably translate in a version bump

**Comment by baywet** [comment]
for context https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41655

**Comment by betlgtu** [comment]
@baywet  "One thing for sure, the addition of static in ClientConstants might break people already referencing it. This should probably translate in a version bump"
Users who use instance methods of ClientConstants will think that everything is ok because build will be successful, but they will recieve runtime JS-error

Methods must be static, not instance

**Comment by baywet** [comment]
so the argument is "it's ok to break it because we're actually fixing it and providing earlier feedback in the loop"? I'm ok with that :)

**Comment by betlgtu** [comment]
Hi everyone,

What I should do to merge PR to master? PR could not be merged because it not reviewed. Nobody wants to look PR.

**Review by baywet — APPROVED** [review]

**Comment by baywet** [comment]
Just approved it because of the time it's been opened.

File: types/sharepoint/tslint.json [hunk 1]
```diff
@@ -18,6 +17,7 @@
         "no-unnecessary-qualifier": false,
         "no-unnecessary-type-assertion": false,
         "prefer-template": false,
-        "use-default-type-parameter": false
+        "use-default-type-parameter": false,
```
Comment (uniqueiniquity) [inline types/sharepoint/tslint.json]: Instead of globally disabling these two lint rules, could you please disable them on a line-by-line basis?

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/sharepoint/tslint.json (hunk 1 above)
Comment (betlgtu) [reply types/sharepoint/tslint.json]: Done

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**